datalad/distributed/create_sibling_gitlab.py

```diff
@@ -412,15 +427,15 @@ def _proc_dataset(refds, ds, site, project, remotename, layout, existing,
             # the reference dataset configuration
             rproject = ds.pathobj.relative_to(refds.pathobj).as_posix()
             if layout == 'hierarchy':
-                project = '{}/{}/_repo_'.format(ref_project, rproject)
+                project = f'{ref_project}/{rproject}/{project_stub}'
             elif layout == 'collection':
                 project = '{}/{}'.format(
                     ref_project,
-                    rproject.replace('/', '--'))
+                    rproject.replace('/', pathsep))
             else:
-                project = '{}--{}'.format(
+                project = '{}-{}'.format(
```

Comment:
```suggestion
                project = '{}{pathsep}{}'.format(
```

Not a change to be applied directly, but wouldn't it make more sense to use `pathsep` here too?

Reply:
I agree. This would address the question I had below in https://github.com/datalad/datalad/pull/7407#pullrequestreview-1468323877. I've tested it locally and it adds the separator in  all the places I expect it to be, i.e. to separate both directories and subdatasets from their parents.

Reply:
Makes sense to me! I think a test needs adjustments, but I'll commit the suggestion and then amend the test fix